src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataReaderExtensions.cs

```diff
@@ -71,7 +72,45 @@ public static ODataItemBase ReadResourceOrResourceSet(this ODataReader reader)
                             "The resource which is ending should be on the top of the items stack.");
                         itemsStack.Pop();
                         break;
+                    case ODataReaderState.DeletedResourceStart:
+                        ODataDeletedResource deletedResource = (ODataDeletedResource)reader.Item;
+                        ODataResourceWrapper deletedResourceWrapper = null;
+                        if (deletedResource != null)
+                        {
+                            deletedResourceWrapper = new ODataResourceWrapper(deletedResource);
+                        }
```

Comment:
deletedResource should never be null if state is ODataReaderState.DeletedResourceStart.